openghg/store/_flux_timeseries.py

```diff
@@ -0,0 +1,256 @@
+            "database_version": database_version,
+            "model": model,
+        }
+
+        signature = inspect.signature(parser_fn)
+        fn_accepted_parameters = [param.name for param in signature.parameters.values()]
+
+        input_parameters: dict[Any, Any] = param.copy()
+
+        # Checks if optional parameters are present in function call and includes them else ignores its inclusion in input_parameters.
+        for param, param_value in optional_keywords.items():
+            if param in fn_accepted_parameters:
+                input_parameters[param] = param_value
+            else:
+                logger.warning(
+                    f"Input: '{param}' (value: {param_value}) is not being used as part of the standardisation process."
+                    f"This is not accepted by the current standardisation function: {parser_fn}"
+                )
+
+        flux_timeseries_data = parser_fn(**input_parameters)
+
+        # Checking against expected format for Flux
+        for split_data in flux_timeseries_data.values():
+            em_data = split_data["data"]
+            FluxTimeseries.validate_data(em_data)
+
+        min_required = ["species", "source"]
+        for key, value in optional_keywords.items():
+            if value is not None:
+                min_required.append(key)
+
+        required = tuple(min_required)
+
+        data_type = "flux_timeseries"
+        datasource_uuids = self.assign_data(
+            data=flux_timeseries_data,
+            if_exists=if_exists,
+            new_version=new_version,
+            data_type=data_type,
+            required_keys=required,
+            compressor=compressor,
+            filters=filters,
+        )
+
+        # Record the file hash in case we see this file again
+        self.store_hashes(unseen_hashes)
+
+        return datasource_uuids
+
+    @staticmethod
+    def validate_data(data: Dataset) -> None:
+        """
+        Validate input data against FluxTimeseries schema - definition from
+        FluxTimeseries.schema() method.
+
+        Args:
+            data : xarray Dataset in expected format
+
+        Returns:
+            None
+
+            Raises a ValueError with details if the input data does not adhere
+            to the FluxTimeseries schema.
+        """
+        data_schema = FluxTimeseries.schema()
+        data_schema.validate_data(data)
+
+    @staticmethod
+    def schema() -> DataSchema:
+        """
+        Define schema for one dimensional timeseries(FluxTimeseries) Dataset.
+
+        Includes observation for each time of the defined domain:
+            - "Obs"
+                - expected dimensions: ("time")
+
+        Expected data types for all variables and coordinates also included.
+
+        Returns:
+            DataSchema : Contains schema for FluxTimeseries.
+        """
+        from openghg.store import DataSchema
+
+        data_vars: Dict[str, Tuple[str, ...]] = {"flux": ("time",)}
```

Comment:
@brendan-m-murphy - can you recall the discussion on this? For the 1D flux timeseries with (time,) dimension would we want this to have the same variable naming as for the 3D flux data with (time, lat, lon). We can do so but this could potentially cause confusion.